src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLanguageServerClient.cs

```diff
@@ -92,9 +121,64 @@ public Task OnServerInitializeFailedAsync(Exception e)
 
         public Task OnServerInitializedAsync()
         {
+            _serverShutdownDisposable = _server.Shutdown.Subscribe((_) => ServerShutdown());
+
+            ServerStarted();
+
             return Task.CompletedTask;
         }
 
+        private void ServerStarted()
```

Comment:
Is there going to be a VSCode equivalent to all this?

Reply:
I actually thought about that and we could absolutely do that; however, it's not super important if we don't want. Reason being VSCode operates on the idea that everything is inside of the workspace directory. 

That being said I know a lot of our users want the ability to open complex projects in that IDE so it's something we could stitch together by grabbing the information on the O# side, publishing it via either VSCode commands or their event pump and then pushing it up to ur server.